pydantic-core/src/validators/mod.rs

```diff
@@ -566,7 +542,20 @@ fn build_validator_inner(
         }
     }
 
-    validator_match!(
+    // macro to build the match statement for validator selection
+    macro_rules! validator_match {
+        ($type:ident, $dict:ident, $config:ident, $definitions:ident, $($validator:path,)+) => {
+            match $type {
+                $(
+                    <$validator>::EXPECTED_TYPE => <$validator>::build($dict, $config, $definitions),
+                )+
+                "invalid" => return Err(invalid_schema_type()),
+                _ => return Err(unknown_schema_type($type)),
+            }
+        };
+    }
```

Comment:
Inlining this code was helpful because it removed the `build_specific_validator` generic function from stack traces so I could see exactly which validator had the `build` method called.